src/main/sensors/opticalflow.c

```diff
@@ -0,0 +1,206 @@
+        return false;
+    }
+
+    opticalflow.dev.init(&opticalflow.dev);
+    opticalflow.quality = OPTICALFLOW_NO_NEW_DATA;
+    opticalflow.rawFlowRates.X = 0;
+    opticalflow.rawFlowRates.Y = 0;
+    opticalflow.processedFlowRates.X = 0;
+    opticalflow.processedFlowRates.Y = 0;
+    opticalflow.lastValidResponseTimeMs = millis();
+
+    return true;
+}
+
+void opticalflowUpdate(void) {
+    if (opticalflow.dev.update) {
+        opticalflow.dev.update(&opticalflow.dev);
+    }
+}
+
+void opticalflowProcess(void) {
+    opticalflowData_t * data = opticalflow.dev.read(&opticalflow.dev);
+    
+    opticalflow.quality   = data->quality;
+
+    if (opticalflow.quality > QUALITY_MINIMUM_THRESHOLD) {
+        opticalflow.lastValidResponseTimeMs = millis();
+
+        opticalflow.rawFlowRates = data->flowRate;
+        opticalflow.deltaTimeUs  = data->deltaTimeUs;
+        
+        applySensorRotation(&opticalflow.processedFlowRates, &opticalflow.rawFlowRates);
+
+        applyLPF(&opticalflow.processedFlowRates);
+
+        // DEBUG SECTION
+        DEBUG_SET(DEBUG_OPTICALFLOW, 0, opticalflow.quality);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 1, opticalflow.rawFlowRates.X);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 2, opticalflow.rawFlowRates.Y);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 3, opticalflow.processedFlowRates.X);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 4, opticalflow.processedFlowRates.Y);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 5, (int32_t)opticalflow.deltaTimeUs);
+    }
+}
+
+static void applySensorRotation(opticalflowRates_t * dist, opticalflowRates_t * src) {
+    if (opticalflowConfig()->rotation != 0) {
+        float angle = DEGREES_TO_RADIANS(opticalflowConfig()->rotation);
+
+        dist->X = (int32_t)(src->X * cosf(angle) - src->Y * sinf(angle));
+        dist->Y = (int32_t)(src->X * sinf(angle) + src->Y * cosf(angle));
+    } else {
+        dist->X = src->X;
+        dist->Y = src->Y;
+    }
+
+    if (opticalflowConfig()->flipX) {
+        dist->X = -dist->X;
+    }
+    
+    if (opticalflowConfig()->flipY) {
+        dist->Y = -dist->Y;
+    }
+}
+
+static void applyLPF(opticalflowRates_t * flowRates) {
+    if (opticalflowConfig()->flow_lpf == 0) {
+        return;
+    }
+    static bool firstRun = true;
+    static pt2Filter_t xFlowLpf, yFlowLpf;
+
+    if (firstRun) {   
+        const float flowCutoffHz = (float)opticalflowConfig()->flow_lpf / 100.0f;
+        const float flowGain     = pt2FilterGain(flowCutoffHz, opticalflow.dev.delayMs / 1000.0f);
+
+        pt2FilterInit(&xFlowLpf, flowGain);
+        pt2FilterInit(&yFlowLpf, flowGain);
+        firstRun = false;
+    }
+
+    flowRates->X = pt2FilterApply(&xFlowLpf, flowRates->X);
+    flowRates->Y = pt2FilterApply(&yFlowLpf, flowRates->Y);
+}
+
+opticalflow_t * getLatestFlowOpticalflowData(void) {
+    return &opticalflow;
+}
+
+bool opticalflowIsHealthy(void) {
+    return (millis() - opticalflow.lastValidResponseTimeMs) < OPTICALFLOW_HARDWARE_TIMEOUT_MS;
```

Comment:
(your code is correct, but this is slightly easier to read and it is more obvious that 32bit overflow is handled correctly:)
```suggestion
    return cmp32(millis(), opticalflow.lastValidResponseTimeMs) < OPTICALFLOW_HARDWARE_TIMEOUT_MS;
```